src/Training.TruckWorld.Clone.Backend/TruckWorld.Infrastructure/Common/Validators/UserValidator.cs

```diff
@@ -0,0 +1,33 @@
+﻿using FluentValidation;
+using Microsoft.Extensions.Options;
+
+using TruckWorld.Application.Common.Settings;
+using TruckWorld.Domain.Entities;
+
+namespace TruckWorld.Infrastructure.Common.Validators;
+
+/// <summary>
+/// Validator class for validating user data using FluentValidation.
+/// </summary>
+public class UserValidator : AbstractValidator<User>
+{
+    /// <summary>
+    /// The validation settings used for user data validation.
+    /// </summary>
+    /// <param name="validationSettings"></param>
+    public UserValidator(IOptions<ValidationSettings> validationSettings)
+    {
+        var validationSettingsValue = validationSettings.Value;
+
+        RuleFor(user => user.EmailAddress)
+            .NotEmpty()
+            .MinimumLength(5)
+            .MaximumLength(64)
+            .Matches(validationSettingsValue.EmailRegexPattern);
+
+        RuleFor(user => user.Password)
+            .NotEmpty()
+            .MinimumLength(8)
+            .MaximumLength(128);
+    }
```

Comment:
We need to change this to rule sets

1. Create enum `EntityEvent` as this in `Domain/Enums`

![image](https://github.com/sulton-max/Training.TruckWorld.Clone/assets/63746735/a23a47ed-ff4e-4c76-bf35-d2c854ea7a7a)

2. Create `RuleSets` as in this example 

![image](https://github.com/sulton-max/Training.TruckWorld.Clone/assets/63746735/cb269cc4-505e-4980-9161-4d7264fe6b0e)

for now rule set for OnCreate is enough, we will add others in separate PR, this PR must be closed quickly